frontend/src/app/modules/players/players-selectors.js

```diff
@@ -62,3 +62,18 @@ const getPlayerCriteriaPoints = (player, matchesLast, criteria) => {
       return player.rating
   }
 }
+
+export const getRankingsForPlayer = createSelector(
+  getTopPlayers,
+  (state, playerId) => playerId,
+  (players, playerId) => getRatingStatisticsForPlayer(players, playerId),
+)
+
+const getRatingStatisticsForPlayer = (players, playerId) => {
+  const index = players.findIndex(player => player.id === playerId)
+  return {
+    ranking: index + 1,
+    toNextRank: index > 0 ? players[index - 1].rating - players[index].rating : 0,
```

Comment:
rank is position detoned by the `getTopRatedPlayers` which is also used in players listing (surrounded by all thhis stuff from filters). Therefore, regardless of the rating there is always someone that will be listed first, second, etc...

Reply:
I think Dashboard and Leaderboard are using getTopPlayers, not getTopRatedPlayers, isn't it? that's why I asked because getTopRatedPlayers uses only ratings if I understand the code right. but even with criteria points - it could be that nextRank will show 0 if ratings are the same, right? that could be confusing...

Reply:
In case of default filters. there is no functional difference between `getTopPlayers` and `getTopRatedPlayers`. Yes, Leaderboard and Dashboard use the same component that uses `getTopPlayers`. The situation where someone will see value of `nextRank` or `prevRank` 0 will happen and is IMHO fine as it describes "The difference between your and the next palyer's score is...". It might be that the label's text does not say this correctly. I can either change label's text or make scores lower/higher by  to match actual label's text, what do you prefer?